library/Icingadb/Common/LoadMore.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/* Icinga DB Web | (c) 2021 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Common;
+
+use Icinga\Module\Icingadb\Widget\ItemList\PageSeparatorItem;
+use Icinga\Module\Icingadb\Widget\ShowMore;
+use ipl\Web\Url;
+
+trait LoadMore
+{
+    /** @var int */
+    protected $pageSize;
+
+    /** @var int */
+    protected $pageNumber;
+
+    /** @var Url */
+    protected $loadMoreUrl;
+
+    /**
+     * Set the page size
+     *
+     * @param int $size
+     *
+     * @return $this
+     */
+    public function setPageSize($size)
+    {
+        $this->pageSize = $size;
+
+        return $this;
+    }
+
+    /**
+     * Set the page number
+     *
+     * @param int $number
+     *
+     * @return $this
+     */
+    public function setPageNumber($number)
+    {
+        $this->pageNumber = $number;
+
+        return $this;
+    }
+
+    /**
+     * Set the page number
+     *
+     * @param Url $url
+     *
+     * @return $this
+     */
+    public function setLoadMoreUrl(Url $url)
+    {
+        $this->loadMoreUrl = $url;
+
+        return $this;
+    }
+
+    protected function getIterator($result)
```

Comment:
* Add phpdoc
* Require that `$result` is a `ResultSet`
  * Then override `__construct` in `HistoryList` and `NotificationList` to also require there that `$data` is a `ResultSet` (but then just call the parent constructor, it's just about the type declaration)

Reply:
Also add the type declaration for the parameter in the method interface.